core/server/proxy/src/main/java/alluxio/proxy/s3/S3ObjectTask.java

```diff
@@ -316,19 +317,25 @@ public Response continueTask() {
             URIStatus status = userFs.getStatus(objectUri);
             FileInStream is = userFs.openFile(status, OpenFilePOptions.getDefaultInstance());
             S3RangeSpec s3Range = S3RangeSpec.Factory.create(range);
-            RangeFileInStream ris = RangeFileInStream.Factory.create(
-                is, status.getLength(), s3Range);
-
-            InputStream inputStream;
+            InputStream inputStream = null;
+            long read = s3Range.getLength(status.getLength());
+            if (read < S3Handler.USE_POSITION_READ_SIZE) {
+              byte[] bytes = new byte[(int) read];
```

Comment:
Can you add a comment here to explain why this if branch is only used in specific size?